src/java.base/share/classes/java/lang/String.java

```diff
@@ -2177,10 +2177,14 @@ public boolean regionMatches(int toffset, String other, int ooffset, int len) {
         if (len <= 0) {
            return true;
         }
+        byte coder = coder();
+        byte otherCoder = other.coder();
         byte[] tv = value;
         byte[] ov = other.value;
-        byte coder = coder();
-        if (coder == other.coder()) {
+        if (coder == otherCoder) {
```

Comment:
Is this change needed as `otherCoder` is not reused?

```suggestion
        byte[] tv = value;
        byte[] ov = other.value;
        byte coder = coder();
        if (coder == other.coder()) {
```

Reply:
Not necessary, but given that later, in the path I have introduced, I am using directly the coder byte to both compute the actual length of `tv` (inlining String::length) and verify the coder to be the same for both `tv` and `ov`, I have avoided fetching it twice.
I see that "could" increase the register pressure, but I haven't verified it yet; there is any performance concern or is a readability/style-only concern?